agent/exec/containerd/adapter.go

```diff
@@ -193,6 +214,29 @@ func (c *containerAdapter) isPrepared() bool {
 	return c.container != nil && c.task != nil
 }
 
+func cniConfig(n *api.Network) (*libcni.NetworkConfig, error) {
+	if n.DriverState.Name != "cni" {
```

Comment:
`cni.ValidateNetworkSpec` would have rejected it if it were, and `cniNetworkAllocator.Allocate` ensures (redundantly) that it is not, but I think I shouldn't rely on the behaviour of such distant components so I will add a check here too.